common/src/OpenSpaceNet.cpp

```diff
@@ -364,29 +370,27 @@ void OpenSpaceNet::processConcurrent()
     recursive_mutex queueMutex;
     deque<pair<cv::Point, cv::Mat>> blockQueue;
     Semaphore haveWork;
-    atomic<bool> cancelled = ATOMIC_VAR_INIT(false);
 
-    MultiProgressDisplay progressDisplay({ "Loading", "Classifying" });
     if(!args_.quiet) {
-        progressDisplay.start();
+        pd_->start();
     }
 
     auto numBlocks = image_->numBlocks().area();
     size_t curBlockRead = 0;
-    image_->setReadFunc([&blockQueue, &queueMutex, &haveWork, &curBlockRead, numBlocks, &progressDisplay](const cv::Point& origin, cv::Mat&& block) -> bool {
+    image_->setReadFunc([&blockQueue, &queueMutex, &haveWork, &curBlockRead, numBlocks, this](const cv::Point& origin, cv::Mat&& block) -> bool {
         {
             lock_guard<recursive_mutex> lock(queueMutex);
             blockQueue.push_front(make_pair(origin, std::move(block)));
         }
 
-        progressDisplay.update(0, (float)++curBlockRead / numBlocks);
+        pd_->update(0, (float)++curBlockRead / numBlocks);
```

Comment:
The magic number that specifies the display index is now really far from when they are defined.

It would be much better to have the progressDisplay be dynamic (as in, you add a new bar right before you need it with a standard label).  Or perhaps semi-dynamic -- have "start" take parameters about what bars to render, instead of setting them up a-priori?

A way to not use a magic number would also be good: such as an enum, etc.